packages/apollo-server-core/src/__tests__/landingPageConfigHTML.test.ts

```diff
@@ -0,0 +1,158 @@
+import {
+  getEmbeddedExplorerHTML,
+  getEmbeddedSandboxHTML,
+} from '../plugin/landingPage/default/index';
+
+const version = '_latest';
+
+describe('Landing Page Config HTML', () => {
+  it('for embedded explorer with document, variables, headers and displayOptions provided', () => {
+    const config = {
```

Comment:
If you define `config` as the actual options type (`ApolloServerPluginEmbeddedLandingPageProductionDefaultOptions`) you won't need the type casts below. Right now TS infers the type of the object (with `docsPanelState` and `theme` as `string`, which is not narrow enough). But if you tell it the object's type, it'll correctly give you an error if you used an incorrect string for those properties.